packages/keystone/fetch.js

```diff
@@ -9,14 +9,33 @@ const logger = getLogger('fetch')
 const FETCH_COUNT_METRIC_NAME = 'fetch.count'
 const FETCH_TIME_METRIC_NAME = 'fetch.time'
 
-async function fetch (url, options) {
+async function fetch (url, options, extraAttrs = {}) {
+
+    const { skipSettingHeaders } = extraAttrs
 
     const urlObject = new URL(url)
     const hostname = urlObject.hostname
     const path = urlObject.pathname
 
     const executionContext = getExecutionContext()
     const parentReqId = executionContext.reqId
+    const parentTaskId = executionContext.taskId
+
+    // We want to set special headers to track requests across the microservices:
+    // Client --reqId-> Condo --reqId-> AddressService
+    //                    ^                   ^
+    //                    |                   |
+    //               log reqId           log reqId
+    //
+    if (!skipSettingHeaders) {
+        if (!options.headers) {
+            options.headers = {}
+        }
+
+        options.headers['X-Request-Id'] = parentReqId || parentTaskId
+        options.headers['reqId'] = parentReqId
+        options.headers['taskId'] = parentTaskId
+    }
```

Comment:
What if no parentReqId or parentTaskId was provided?

Reply:
what do you think?

Reply:
I suppose you should set value explicitly. In this case we can filter requests without that prop